space_packet_parser/packets.py

```diff
@@ -97,10 +97,7 @@ def data_length(self) -> int:
         Section 4.1.3.5.3 The length count C shall be expressed as:
         C = (Total Number of Octets in the Packet Data Field) – 1
         """
-        # This has already been parsed previously to give us the length of the packet
-        # so avoid the extract_bits call again and calculate it based on the length of the data
-        # Subtract 6 bytes for the header and 1 for the length count
-        return len(self) - RawPacketData.HEADER_LENGTH_BYTES - 1
+        return (self[4] << 8) | self[5] + 1
```

Comment:
I wasn't sure which value we want to return here. The "C" (value inside the packet) or "C + 1" (the true length of the data field). I can see arguments for either of these values.

Reply:
I think we want the CCSDS defined value, C because that's the value encoded in the field that this is accessing, following the pattern of the other property accessors. If someone wants the total length of the packet, that's `len(packet_data)`. If they want the total length minus the header, that's `len(packet_data) - RawPacketData.HEADER_LENGTH_BYTES` or `packet_data.data_length + 1`